### PR TITLE
[rfxcom] Add ability to properly receive configured command ids, deprecate hard-coded guesses

### PR DESCRIPTION
**Pull request by Jamstah**
Currently, when a message is received the command will be determined only from the hard-coded set of ON/OFF commands, which means if I configure a thing and attach it to a switch there is no guarantee that it will respond as expected to receive commands.

The hard-coded ON/OFF commands are based on experimentation with different devices, and are not at all consistent. This commit also deprecates the use of those so that in a future release, Lighting4 devices will only work when configured with the correct command ids up front. This will resolve inconsistent behaviour where devices that have been discovered may work correctly, work only for ON or OFF but not both, have ON and OFF the wrong way around, turn ON one physical device and OFF another, or any combination of those possibilities.

Signed-off-by: James Hewitt <james.hewitt@uk.ibm.com>

### CODE REVIEW

**Comment by Jamstah** [comment]
Making draft for now, forgot to clean up how messages are matched to things.

File: bundles/org.openhab.binding.rfxcom/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComUndecodedRFMessageTest.java [hunk 1]
```diff
@@ -51,12 +50,21 @@ public void testSomeMessages() throws RFXComException {
...
-        msg.subType = RFXComUndecodedRFMessage.SubType.ARC;
-        msg.seqNbr = 1;
-        msg.rawPayload = HexUtils.hexToBytes("000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F2021");
+        assertThrows(RFXComMessageTooLongException.class,
+                () -> testMessage("25030101000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F2021",
+                        RFXComUndecodedRFMessage.SubType.ARC, 0x01,
+                        "000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F2021"));
 
-        assertThrows(RFXComMessageTooLongException.class, () -> msg.decodeMessage());
+        // RFXComGenericDeviceConfiguration config = new RFXComGenericDeviceConfiguration();
```
Comment (martinvw) [inline bundles/org.openhab.binding.rfxcom/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComUndecodedRFMessageTest.java:58]: Should this be removed or commented out?

Comment (Jamstah) [reply bundles/org.openhab.binding.rfxcom/src/test/java/org/openhab/binding/rfxcom/internal/messages/RFXComUndecodedRFMessageTest.java:58]: Removed, thanks :)

**Comment by Jamstah** [comment]
Glad I set to draft, because this wasn't setting the pulse and was crashing the hardware. Now fixed and some extra tests added.

This is now ready for review.

Once (if) this goes in, I'll add a suitable note to https://github.com/openhab/openhab-distro/blob/main/distributions/openhab/src/main/resources/bin/update.lst.

**Review by martinvw — APPROVED** [review]

**Comment by martinvw** [comment]
@Jamstah should this be marked as a bug or enhancement?

**Comment by Jamstah** [comment]
Marked where?

I'd say enhancement

**Pull request merged**